### PR TITLE
Feature site ui

### PR DESCRIPTION
**Pull request by duninvit**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -24,47 +24,49 @@ apply plugin: 'org.springframework.boot'
...
-	compile('org.springframework.security:spring-security-jwt:1.0.4.RELEASE')
-	compile('com.fasterxml.jackson.module:jackson-module-kotlin')
-	compile("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
-	compile("org.jetbrains.kotlin:kotlin-reflect")
-	runtime('org.postgresql:postgresql')
-	compileOnly('org.projectlombok:lombok')
+    compile('org.springframework.boot:spring-boot-starter-security')
+    compile('org.springframework.boot:spring-boot-starter-web')
+    compile('org.springframework.boot:spring-boot-starter-data-jpa')
+    compile('org.springframework:spring-websocket:5.0.6.RELEASE')
```
Comment (MaximYesman) [inline build.gradle]: remove versions

File: site-ui/package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "test": "ng test",
+    "lint": "ng lint",
+    "e2e": "ng e2e"
+  },
+  "private": true,
+  "dependencies": {
+    "@angular/animations": "^6.0.5",
```
Comment (MaximYesman) [inline site-ui/package.json]: use fixed versions. Do not fix now, just note this

File: site-ui/src/app/components/layout/chat/chat.component.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+@Component({
+  selector: 'app-chat',
+  templateUrl: './chat.component.html',
+  styleUrls: ['./chat.component.css'],
+})
+
+export class ChatComponent implements OnDestroy {
+
+  greetings: {
```
Comment (MaximYesman) [inline site-ui/src/app/components/layout/chat/chat.component.ts]: create dto

File: site-ui/src/app/components/layout/chat/chat.component.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    private route: ActivatedRoute,
+    private router: Router,
+    private webSocketService: WebSocketService
+  ) {
+    this.navigationSubscription = this.router.events.subscribe((e: any) => {
+      if (e instanceof NavigationStart) {
+        this.Destroy();
+      }
+    });
+    this.navigationSubscription = this.router.events.subscribe((e: any) => {
```
Comment (MaximYesman) [inline site-ui/src/app/components/layout/chat/chat.component.ts]: can be in single subscription

File: site-ui/src/app/components/layout/chat/chat.component.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  navigationSubscription;
+
+  constructor(
+    private tokenStorageService: TokenStorageService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private webSocketService: WebSocketService
+  ) {
+    this.navigationSubscription = this.router.events.subscribe((e: any) => {
```
Comment (MaximYesman) [inline site-ui/src/app/components/layout/chat/chat.component.ts]: put in OnInit method

File: site-ui/src/app/components/layout/chat/chat.component.ts (hunk 4 above)
Comment (MaximYesman) [inline site-ui/src/app/components/layout/chat/chat.component.ts]: reassign

File: site-ui/src/app/components/layout/chat/chat.component.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        this.Init();
+      }
+    });
+  }
+
+  Init() {
+    this.greetings = [];
+    this.chatId = this.route.snapshot.params['id'];
+    const that = this;
+    this.webSocketService.subscribe(this.route.snapshot.params['id'], function (message) {
```
Comment (MaximYesman) [inline site-ui/src/app/components/layout/chat/chat.component.ts]: **function (message) {** can be replaced with **message => {**

File: site-ui/src/app/components/layout/panel/panel.component.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-panel',
+  templateUrl: './panel.component.html',
+  styleUrls: ['./panel.component.css']
+})
+export class PanelComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
```
Comment (MaximYesman) [inline site-ui/src/app/components/layout/panel/panel.component.ts]: redundant

File: site-ui/src/app/components/login/login.component.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
```
Comment (MaximYesman) [inline site-ui/src/app/components/login/login.component.ts]: redundant

File: site-ui/src/app/components/signup/signup.component.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-signup',
+  templateUrl: './signup.component.html',
+  styleUrls: ['./signup.component.css']
+})
+export class SignupComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
```
Comment (MaximYesman) [inline site-ui/src/app/components/signup/signup.component.ts]: redundant

File: src/main/kotlin/com/example/site/service/ChatService.kt [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+@Transactional
+class ChatService(private val messageRepository: MessageRepository,
+                  private val roomRepository: RoomRepository,
+                  private val userRepository: UserRepository) {
+
+    fun sendMessage(messageDto: MessageDto, roomId: Long): MessageDto {
+        val room = roomRepository.findById(roomId).orElseThrow { throw Exception() }
```
Comment (MaximYesman) [inline src/main/kotlin/com/example/site/service/ChatService.kt]: throw something else

Comment (MaximYesman) [inline src/main/kotlin/com/example/site/service/ChatService.kt]: write extension method to throw anywhere same exception

File: src/main/kotlin/com/example/site/service/ChatService.kt [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+        if (room.type == RoomType.SAFE) {
```
Comment (MaximYesman) [inline src/main/kotlin/com/example/site/service/ChatService.kt]: write separate handlers for all room types

File: src/main/kotlin/com/example/site/service/ChatService.kt [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+            val user = userRepository.findById(messageDto.owner).orElseThrow { throw Exception() }
```
Comment (MaximYesman) [inline src/main/kotlin/com/example/site/service/ChatService.kt]: throw other

File: site-ui/src/app/components/layout/chat/chat.component.ts (hunk 5 above)
Comment (duninvit) [reply site-ui/src/app/components/layout/chat/chat.component.ts]: don't work in OnInit

File: site-ui/src/app/components/login/login.component.ts (hunk 8 above)
Comment (duninvit) [reply site-ui/src/app/components/login/login.component.ts]: TODO

File: site-ui/src/app/components/layout/chat/chat.component.ts [hunk 13]
```diff
@@ -76,3 +71,8 @@ export class ChatComponent implements OnDestroy {
     this.greetings.push(JSON.parse(message));
   }
 }
+
+export class Greetings {
```
Comment (MaximYesman) [inline site-ui/src/app/components/layout/chat/chat.component.ts]: move to separate file/folder

File: src/main/kotlin/com/example/site/service/ChatUnsafeHandler.kt [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.stereotype.Component
+
+@Component
+class ChatUnsafeHandler(private val messageRepository: MessageRepository,
+                        private val userRepository: UserRepository) : RoomHandler {
+
+    override val roomType: RoomType = RoomType.UNSAFE
+
+    override fun process(messageDto: MessageDto, room: Room) {
+        val user = userRepository.findById(messageDto.owner).orElseThrow { throw JsonException("User not found") }
```
Comment (MaximYesman) [inline src/main/kotlin/com/example/site/service/ChatUnsafeHandler.kt]: **orElseThrow { throw JsonException("User not found") }** - to extension method on Optional<T>

**Pull request merged**